tests/phpunit/tests/admin/wpAutomaticUpdater.php

```diff
@@ -69,7 +69,7 @@ function( $args ) use ( $urls ) {
 		$has_failed     = ! empty( $failed );
 
 		if ( ! $has_successful && ! $has_failed ) {
-			$this->markTestSkipped( 'This test requires at least one successful or failed plugin update object.' );
+			$this->fail( 'This test requires at least one successful or failed plugin update object.' );
```

Comment:
As far as I can see, this should be handled via a [`@depends`](https://docs.phpunit.de/en/9.6/annotations.html#depends) tag, not via test skipping/failing.

Tests are not guaranteed to run in the same order and can even be explicitly requested to be run in a different order [using `--order-by`](https://docs.phpunit.de/en/9.6/textui.html#command-line-options). This test could now cause random test failures based on the test execution order.

As a side-note: running tests in random order is a great way to find tests which are too closely coupled or don't do their setup/teardown correctly.